### PR TITLE
Deprecate support for `ember-cli-jshint`

### LINKED ISSUE #9882: Remove support for `ember-cli-jshint`?
`ember-cli-jshint` was removed a long time ago, but it seems we still have a few parts in the codebase that have checks for this:

https://github.com/ember-cli/ember-cli/blob/b24b73b388934796ca915ca665b48a27c857199b/blueprints/blueprint/index.js#L6-L20

https://github.com/ember-cli/ember-cli/blob/b24b73b388934796ca915ca665b48a27c857199b/blueprints/server/index.js#L30-L38

### PR DESCRIPTION
**Pull request by bertdeblock**
Closes #9882.

### CODE REVIEW

File: blueprints/blueprint/index.js [hunk 1]
```diff
@@ -2,20 +2,4 @@
 
 module.exports = {
   description: 'Generates a blueprint and definition.',
-
-  files() {
```
Comment (bertdeblock) [inline blueprints/blueprint/index.js:6]: The `blueprint` blueprint doesn't contain a `.jshintrc` file anymore, so this `files` logic was redundant.

File: tests/unit/blueprints/lib-test.js [hunk 2]
```diff
@@ -5,7 +5,6 @@ const blueprintHelpers = require('ember-cli-blueprint-test-helpers/helpers');
 let setupTestHooks = blueprintHelpers.setupTestHooks;
```
Comment (bertdeblock) [inline tests/unit/blueprints/lib-test.js:5]: The `blueprint` blueprint doesn't contain the `.jshintrc` and `.eslintrc.js` files anymore, hence the removed code in this test.

**Review by kellyselden — APPROVED** [review]

File: blueprints/server/index.js [hunk 3]
```diff
@@ -32,8 +33,18 @@ module.exports = {
...
-    if (this.project) {
-      return 'ember-cli-jshint' in this.project.dependencies();
-    }
+    let hasJSHint = Boolean(this.project) && 'ember-cli-jshint' in this.project.dependencies();
+
+    deprecate('Support for `ember-cli-jshint` is deprecated. We recommend using `eslint` instead.', !hasJSHint, {
+      for: 'ember-cli',
+      id: 'ember-cli.ember-cli-jshint-support',
+      since: {
+        available: '4.5.0',
```
Comment (kellyselden) [inline blueprints/server/index.js:42]: I think this would be 4.6.0 at this point, unless you targeted beta.

Comment (kellyselden) [reply blueprints/server/index.js:42]: I might be wrong since we haven't done 4.4 yet.

Comment (bertdeblock) [reply blueprints/server/index.js]: Updated to `4.6.0` since `4.4.0` is now released.

**Review by kellyselden — APPROVED** [review]

**Pull request merged**